File: codeforces_805/c.cpp
#include <bits/stdc++.h>

using namespace std;
 
 
void solve(void) {
    int n, k;
    map<int, int> min_indices;
    map<int, int> max_indices;

    cin >> n >> k;

    vector<int> stations(n);
    for (int j=0; j < n; j++) {
        cin >> stations[j];
    }

    for (int j = 0; j < n; j++) {
        if (min_indices.find(stations[j]) == min_indices.end()) {
            min_indices[stations[j]] = j;
        }
        max_indices[stations[j]] = j;
    }

    while (k--) {
        int start, end;

        cin >> start >> end;

        if (max_indices.find(end) == max_indices.end() || min_indices.find(start) == min_indices.end()) {
            cout << "NO" << endl;
            continue;
        }

        if (max_indices[end] <= min_indices[start]) {
            cout << "NO" << endl;
        }
        else {
            cout << "YES" << endl;
        }

    }

}
 
int main(void) {
    int t;
    cin >> t;

    while (t--) {
        solve();
    };
 
    return 0;
 
};
 

File: codeforces_805/c.py
from sys import stdin
 
 
def solve():
    skip = input()
    # create an indices hashmap and compare the target to source
    n, k = list(map(int, stdin.readline().rstrip().split()))
 
    stations = list(map(int, stdin.readline().rstrip().split()))
    min_indices = {}
    max_indices = {}
    
    for ind, station in enumerate(stations):
        if station not in min_indices:
            min_indices[station] = ind
        max_indices[station] = ind
 
    for query in range(k):
        not_possible = False

        start, end = list(map(int, stdin.readline().rstrip().split()))
 
        if end not in max_indices or start not in min_indices:
            not_possible = True
        
        if not_possible:
            print("NO")
        else:
            end_index = max_indices[end]
            start_index = min_indices[start]
     
            if end_index <= start_index:
                print("NO")
            else:
                print("YES")
 
 
def main():
    t = int(input())
 
    for i in range(t):
        solve()
 
    return 0
 
 
main()
